internal/common/function/function_test.go

```diff
@@ -688,6 +693,241 @@ func TestValidateRestoreSpecEnforcedFields(t *testing.T) {
 	})
 }
 
+func TestValidateBslSpec(t *testing.T) {
```

Comment:
would follow structure of NAB and NAR tests https://github.com/migtools/oadp-non-admin/blob/019bb2a677d4648cce1443966d6686cd46a7cef4/internal/common/function/function_test.go#L94-L377

Reply:
I don't understand this request. The difference here is that at the beginning we need to register new corev1 schema and then we create the `tests := [] struct {}` which is pretty much similar to NAB and NAR ?

Reply:
for NAB and NAR the test is split in 2: one for enforcement fields and one for other validations

for enforcement, we also check that every single field was tested https://github.com/migtools/oadp-non-admin/blob/019bb2a677d4648cce1443966d6686cd46a7cef4/internal/common/function/function_test.go#L361

Reply:
Thanks for explanation, will look into that.